### PR TITLE
[ARRDT] Add support for strict global object id store keys

### PR DESCRIPTION
**Pull request by alloy**

### CODE REVIEW

**Review by vladar — APPROVED** [review]
Looks good to me. But see one question inline.

File: packages/apollo-react-relay-duct-tape/src/storeObservation/nodeFromCacheFieldPolicy.test.ts [hunk 1]
```diff
@@ -27,55 +34,97 @@ const MOCK_DATA = {
...
+  describe("with strict Global Object Identification spec store keys", () => {
+    beforeAll(() => {
+      client = new ApolloClient({
+        cache: new InMemoryCache({
+          dataIdFromObject(responseObject) {
+            return (
+              responseObject.id?.toString() ||
+              defaultDataIdFromObject(responseObject)
+            );
+          },
```
[the remark this answers is not in the document]
Comment (alloy) [reply packages/apollo-react-relay-duct-tape/src/storeObservation/nodeFromCacheFieldPolicy.test.ts:110]: Officially the `defaultDataIdFromObject` function is already deprecated and the implementation is simple enough, so I'd rather not tuck it away in our lib and rather have people look at the docs https://www.apollographql.com/docs/react/caching/cache-configuration/

[the remark this answers is not in the document]
Comment (alloy) [reply packages/apollo-react-relay-duct-tape/src/storeObservation/nodeFromCacheFieldPolicy.test.ts:110]: What I _should_ do, however, is add doc to the field policy function.

[the remark this answers is not in the document]
Comment (alloy) [reply packages/apollo-react-relay-duct-tape/src/storeObservation/nodeFromCacheFieldPolicy.test.ts:110]: Done

**Pull request merged**